crates/client/src/telemetry.rs

```diff
@@ -341,6 +341,13 @@ impl Telemetry {
             .detach();
     }
 
+    pub fn metrics_enabled(self: &Arc<Self>) -> bool {
+        let state = self.state.lock();
+        let enabled = state.settings.metrics;
+        drop(state);
+        return enabled;
```

Comment:
We shouldn't need an explicit `return` here:

```suggestion
        enabled
```